server/servercontext.py

```diff
@@ -53,24 +54,59 @@ def close(self):
     def __contains__(self, connection):
         return connection in self.connections.keys()
 
+    async def broadcast(self, message, validate_fn=lambda a: True):
+        await self._do_broadcast(
+            validate_fn,
+            QDataStreamProtocol.send_message,
+            message
+        )
+        self._logger.log(TRACE, "]]: %s", message)
+
     async def broadcast_raw(self, message, validate_fn=lambda a: True):
+        await self._do_broadcast(
+            validate_fn,
+            QDataStreamProtocol.send_raw,
+            message
+        )
+
+    async def _do_broadcast(self, validate_fn, send_fn, message):
         server.stats.incr('server.broadcasts')
+
+        async def broadcast_with_stall_handling(proto):
```

Comment:
Typing + we can move that inner function out of the scope of `_do_broadcast`.
Inner functions are bad practice + hard to test, etc...